### PR TITLE
Signup and signin

### PR DESCRIPTION
**Pull request by Sharmaryan**
- added signup and sigin page
- added authentication

### CODE REVIEW

**Review by khushijohri2001 — APPROVED** [review]

**Comment by khushijohri2001** [comment]
Code looks good to me

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_JWT_SECRET=kuchbhi
```
Comment (ravikumar0403) [inline .env]: don't push .env to github

File: src/App.css [hunk 2]
```diff
@@ -1,6 +1,4 @@
-.App {
-  text-align: center;
-}
+
 
 .App-logo {
```
Comment (ravikumar0403) [inline src/App.css]: You can remove this if its not used

File: src/components/Navbar/Navbar.jsx [hunk 3]
```diff
@@ -1,49 +1,55 @@
-import React from 'react'
-import {Link} from 'react-router-dom'
-import './Navbar.css'
+import React from "react";
+import { Link } from "react-router-dom";
+import "./Navbar.css";
+import { useAuth } from "../../context/auth-context";
```
Comment (ravikumar0403) [inline src/components/Navbar/Navbar.jsx:4]: use path alias to prevent navigating by using `../../`

File: src/components/Navbar/Navbar.jsx [hunk 4]
```diff
@@ -1,49 +1,55 @@
...
+        <i className="fas fa-search search-background"></i>
+        <input
+          type="text"
+          placeholder="search products"
+          className="search-bar"
+        />
+      </div>
+      <div className="menu">
+        <ul>
+          {!auth.auth && (
```
Comment (ravikumar0403) [inline src/components/Navbar/Navbar.jsx:25]: you can destructure and pass from the context

File: src/components/SignUp/SignUpComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+export const SignUpComponent = () => {
+  const signupReducer = (state, { fieldName, fieldValue, type }) => {
+    switch (type) {
+      case "INPUT":
+        return { ...state, [fieldName]: fieldValue };
+      default:
+        return { ...state };
+    }
+  };
+  const [{ firstName, lastName, email, password }, dispatch] = useReducer(
```
Comment (ravikumar0403) [inline src/components/SignUp/SignUpComponent.jsx:9]: you can extract out the reducer function in a separate file

File: src/components/SignUp/SignUpComponent.jsx [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+            />
+          </label>
+
+          <div className="t-and-c">
+            <label className="label" htmlFor="t&c">
+              <input name="t&c" id="t&c" type="checkbox" />I accept terms and
+              condition
+            </label>
+          </div>
+          <button className="signup-btn" onClick={(e) => signupHandler(e)}>
```
Comment (ravikumar0403) [inline src/components/SignUp/SignUpComponent.jsx]: If you pass the function reference then event parameter is passed implicitly to the function

**Review by ravikumar0403 — APPROVED** [review]
Good folder structure. I've left a few comments

**Pull request merged**